logic.typ

```diff
@@ -159,6 +159,50 @@
   }
 }
 
+#let fill-remaining-height(
+  margin: 0%,
+  before-label: "before-fill",
+  ..box-kwargs,
+  h-align: left,
+  content,
+) = {
+  locate(loc => {
+    let prev = query(selector(label("before-fill")).before(loc), loc)
+    if prev.len() == 0 {
+      panic("You must use the label `" + before-label + "` before calling this function")
+    }
+    let prev-pos = prev.last().location().position()
+    let loc-page = loc.position().page
+    if prev-pos.page != loc-page {
+      panic(
+        "Label `" + before-label + "` must appear on the same page as the content passed"
+        + " to this function (page " + str(loc-page) + "), but the last found label"
+        + " was on page " + str(prev-pos.page)
+      )
+    }
+    layout(page-size => {
```

Comment:
`layout` gives us the current container's dimensions, not necessarily those of the page. That is fine for this use case, I think, but we should hence name this `container-size` instead.